packages/langium/src/workspace/documents.ts

```diff
@@ -327,6 +355,22 @@ export class DefaultLangiumDocuments implements LangiumDocuments {
     }
 
     invalidateDocument(uri: URI): void {
+        const uriString = uri.toString();
+        const langiumDoc = this.documentMap.get(uriString);
+        if (langiumDoc) {
+            langiumDoc.state = DocumentState.Changed;
+            langiumDoc.parseResult = {
+                lexerErrors: [],
+                parserErrors: [],
+                value: { // dummy AstNode to have the parseResult properly initialized
+                    $type: 'AstNode',
+                    $document: langiumDoc
+                }
+            };
```

Comment:
Suggestion: I wouldn't put a dummy node here. The user will always have the information, that the document is outdated through the `state` property. Putting a node in here which doesn't fit the constraints of the `LangiumDocument<T>` parameter might lead to runtime errors, if the user tries to access those.

Reply:
Do we state somewhere that client code should check the state before accessing the content a document?
I didn't find a hint on that, by now.

> might lead to runtime errors, if the user tries to access those
That's true. Do you have another proposal?
Just leaving the stale content in there feels also error-prone to me. With EMF we proxified the AST in that case...

Reply:
We should just document that cross-references might not be resolved when you access a document with state < Linked. If you want to do something with a properly linked document, you can hook into the DocumentBuilder phases.